chain-impl-mockchain/src/testing/e2e/transactions.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{
+    fee::LinearFee,
+    testing::{
+        ledger::ConfigBuilder,
+        scenario::{prepare_scenario, wallet},
+        verifiers::LedgerStateVerifier,
+    },
+    value::Value,
+};
+use chain_addr::Discrimination;
+use quickcheck::TestResult;
+use quickcheck_macros::quickcheck;
+use rand::Rng;
+
+#[quickcheck]
+pub fn validate_ledger_state_after_transaction(amount: u64) -> TestResult {
+    if amount == 0 {
+        return TestResult::discard();
+    };
+
+    let mut rng = rand::thread_rng();
```

Comment:
it is not very quickcheckish way of getting random number. I would write a bit more code but it would be nicely incorporated into framework.

First let's create arbitrary for u64 with value constraint (1..10):
```
struct Random1to10(u64);
```
and then implement arbitrary
```
impl Arbitrary for Random1to10 {
     fn arbitrary<G: Gen>(g: &mut G) -> Self {
            Self(u64::arbitrary(g) % 10 + 1)
     }
}
```

Finally use it in quickcheck:

```
pub fn validate_ledger_state_after_transaction(amount: u64, random: Random1to10 ) -> TestResult {
```